src/api/test/functional/request_controller_test.rb

```diff
@@ -2675,8 +2675,8 @@ def test_auto_accept_request
     # the other one got close because the target does not exist anymore
     get "/request/#{id2}"
     assert_response :success
-    assert_xml_tag(tag: 'state', attributes: { name: 'revoked', when: '2010-07-14T00:00:00', who: 'Iggy' })
-    assert_xml_tag(tag: 'comment', content: 'Permission problem')
+    assert_xml_tag(tag: 'state', attributes: { name: 'declined', when: '2010-07-14T00:00:00', who: 'Iggy' })
+    assert_xml_tag(tag: 'comment', content: "The target package 'TestPack' has been removed")
```

Comment:
There is a lot of code handling how bs requests are processed and states are changed. Mainly I wanted to fully understand this before changing the tests.

tldr;

Before there was a permission check running multiple times. This caused an exception that got catched and sets the state to 'revoked' – 'Permission problem'.
Now the check only runs once and the states ends as 'declined' – target package has been removed

--------------

the long version:

The difference before / after this PR is that the [cmd_changestate_permissions](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/bs_request_permission_check.rb#L305) were run 3 times for BsRequest number 1043 (from the test). In one of these runs an [exception](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/bs_request_permission_check.rb#L324) is thrown and then catched in the [auto_accept method](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/bs_request.rb#L935). There it used to be set to revoked because of permission issues.

With the codechange of this PR:
 - permission check runs one time
 - therefore no permission exception
 - bs request ends with state 'declined'